### PR TITLE
add installation directives to cmakelists.txt

### PR DESCRIPTION
**Pull request by Cylix**
This change addresses #297.

@thibault-martinez this will only install iota.lib.cpp includes and library, and the tests binary if they were built.
Do you think we should also install dependencies such as cpr / json / curl?

I think there are two things we may need to consider:

* we do include some libraries in our header files, thus json headers need to be somewhere in the path when building our library.
* curl provides a shared library which is loaded at runtime, so we may need to install it such that it is in the installation path as well.
* installing those may conflict with previously installed libraries.

Do you have any particular opinion on this?
I would be fine with installing those deps as well.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -88,6 +88,17 @@ include(curl_settings)
 include(cpr_settings)
 include(keccak_settings.as_dep)
 
+########## install ##########
+
+# ensure lib and bin directories exist
+install(CODE "FILE(MAKE_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})")
+install(CODE "FILE(MAKE_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})")
+
+# install cpp_redis
```
Comment (thibault-martinez) [inline CMakeLists.txt]: Rename please ;)

Comment (Cylix) [reply CMakeLists.txt]: ahah, good catch :D

**Comment by thibault-martinez** [comment]
@Cylix Same here, please rebase

**Comment by Cylix** [comment]
Not sure why, on appveyor it built IRI 1.5.0 but with the old .appveyor config file.
I rebased master, hopefully should be good.

**Pull request merged**